Adding FMC support to STM32U5

potentialdiffer opened the pull request:
Adding fmc definition to device tree and solved a minor warning.

erwango commented on dts/arm/st/u5/stm32u5.dtsi:
```diff
@@ -149,6 +150,7 @@
 			compatible = "st,stm32-exti";
 			interrupt-controller;
 			#interrupt-cells = <1>;
+			#address-cells = <1>;
```
Thanks for fixing this, but please do it in a dedicated commit (mentioning the warning you fix).
Pushing a bit, even a dedicated PR would be nice so we can discuss more around this, as I'm not seeing this warning on my side, and maybe fix on all series.

potentialdiffer replied:
Sure, will create separate commits.

I can add the configuration for the B-U585I-IOT02A. I just saw, it has additional SRAM available. However, I will not be able to test it myself.

erwango replied:
> On the B-U585I-IOT02A, the SRAM is connected via SPI and not via FMC. The nucleo_u575zi_q also does not have any FMC connected hardware

Ok, then let it aside.

potentialdiffer replied:
Alright, I removed the last commit. I think this request can be resolved now?

erwango replied:
Seems it reappeared somehow

erwango reviewed (changes requested):
Thanks for introducing this.

One request: would you mind enabling support on an in tree board (B-U585I-IOT02A), so it can be easily re-used by others and more importantly tested in CI.

potentialdiffer committed:
dts: stm32: u5: adding st-fmc support to stm32u5

No fmc node for the stm32u5 is implemented. This commit
adds a stm32-fmc compatible node to the device tree.

Signed-off-by: Christian Spinnler <christian.spinnler@fau.de>

FRASTM commented on dts/arm/st/u5/stm32u5.dtsi:
```diff
@@ -662,6 +664,20 @@
+		fmc: memory-controller@420d0400 {
+			compatible = "st,stm32-fmc";
+			reg = <0x420d0400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB3 0x00001000>;
```
can you please confirm, I see the FSMC clock control on the  AHB2_2 bit 0   (AHB3 bit 12 is GTZC)

potentialdiffer replied:
You are right, I double checked in the documentation. I have to say, I just took over the configuration from H7, which did not align with U5. I fixed the clocks definition.

erwango replied:
One question then: did you test it ?

potentialdiffer replied:
Not yet, as the hardware is not yet ready. Will be able to test it next week.

erwango replied:
ok. Then I prefer we don't merge and wait for tests results.
It's possible that FMC on U5 requires more than only dtsi addition. Some drivers changes might be required as well. This is usual on U5 series.

erwango replied:
@potentialdiffer please add test configuration as part of this PR in a dedicated commit

FRASTM commented on dts/arm/st/u5/stm32u5.dtsi:
```diff
@@ -683,6 +685,20 @@
+		fmc: memory-controller@420d0400 {
+			compatible = "st,stm32-fmc";
+			reg = <0x420d0400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB2_2 0x00000000>;
```
```suggestion
			clocks = <&rcc STM32_CLOCK_BUS_AHB2_2 0x00000001>;
```

potentialdiffer replied:
Fixed this

potentialdiffer committed:
tests: drivers: add nor-psram to memc test configuration

This enables running stm32-fmc-nor-psram enabled devices to run memc test

Signed-off-by: Christian Spinnler <christian.spinnler@fau.de>
